lib/elixir/src/elixir_tokenizer.erl

```diff
@@ -609,8 +609,13 @@ handle_strings(T, Line, Column, H, Scope, Tokens) ->
       Token = {Key, {Line, Column - 1, nil}, Unescaped},
       tokenize(Rest, NewLine, NewColumn + 1, Scope, [Token | Tokens]);
     {NewLine, NewColumn, Parts, Rest} ->
-      Token = {string_type(H), {Line, Column - 1, nil}, unescape_tokens(Parts, Scope)},
-      tokenize(Rest, NewLine, NewColumn, Scope, [Token | Tokens])
+      UnescapedToken = unescape_tokens(Parts, Scope),
+      case UnescapedToken of
+        {error, _Msg} = Error -> Error;
```

Comment:
I don't think we should introduce a new error tuple which is {error, String}. Let's change this line so it returns: `{error, {Line, Column, Message, string_type(H)}, Rest, Tokens}`. This way you won't have to change any of the calls to `:elixir.string_to_token` and ultimately the code will be simpler.